sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,377 @@
+    const contentRootElementName = contentElementNames[0];
+    delete entry.content[contentRootElementName][CoreHttpConstants.XML_METADATA_MARKER];
+    result = entry.content[contentRootElementName];
+
+    if (result) {
+      if (entry[CoreHttpConstants.XML_METADATA_MARKER]) {
+        result[ServiceBusConstants.ATOM_METADATA_MARKER] =
+          entry[CoreHttpConstants.XML_METADATA_MARKER];
+      } else {
+        result[ServiceBusConstants.ATOM_METADATA_MARKER] = {};
+      }
+
+      result[ServiceBusConstants.ATOM_METADATA_MARKER][
+        "ContentRootElement"
+      ] = contentRootElementName;
+
+      Object.keys(entry).forEach((property: string) => {
+        if (property !== "content" && property !== CoreHttpConstants.XML_METADATA_MARKER) {
+          result[ServiceBusConstants.ATOM_METADATA_MARKER][property] = entry[property];
+        }
+      });
+
+      return result;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * @ignore
+ * Utility to help parse given `feed` result
+ * @param feed
+ */
+function parseFeedResult(feed: any): object[] {
+  const result = [];
+  if (typeof feed === "object" && feed != null && feed.entry) {
+    if (Array.isArray(feed.entry)) {
+      feed.entry.forEach((entry: any) => {
+        const parsedEntryResult = parseEntryResult(entry);
+        if (parsedEntryResult) {
+          result.push(parsedEntryResult);
+        }
+      });
+    } else {
+      const parsedEntryResult = parseEntryResult(feed.entry);
+      if (parsedEntryResult) {
+        result.push(parsedEntryResult);
+      }
+    }
+  }
+  return result;
+}
+
+function isKnownResponseCode(
+  statusCode: number
+): statusCode is keyof typeof ServiceBusConstants.HttpResponseCodes {
+  return !!(ServiceBusConstants.HttpResponseCodes as { [statusCode: number]: string })[statusCode];
+}
+
+/**
+ * @ignore
+ * Extracts the applicable entity name(s) from the URL based on the known structure
+ * and instantiates the corresponding name properties to the deserialized response
+ *
+ * For instance, following is the URL structure for when creating a rule
+ * `<namespace-component>/<topic-name>/Subscriptions/<subscription-name>/Rules/<rule-name>`
+ *
+ * @param entry
+ * @param nameProperties
+ */
+function setName(entry: any, nameProperties: any): any {
+  if (entry[ServiceBusConstants.ATOM_METADATA_MARKER]) {
+    const parsedUrl = new URL(entry[ServiceBusConstants.ATOM_METADATA_MARKER].id);
+
+    let pathname: string = parsedUrl.pathname;
+
+    // The parsedUrl gets constructed differently for browser vs Node.
+    // It is specifically behaves different for some of the Atom based management API where
+    // the received URL in "id" element is of type "sb:// ... " and not a standard HTTP one
+    if (!isNode) {
+      if (pathname.startsWith("//")) {
+        pathname = pathname.substring(3);
+      }
+    }
+
+    const parts = pathname.split("/");
+
+    for (let i = 0; i * 2 < parts.length - 1; i++) {
```

Comment:
I am a little wary of relying on extracting every alternate value to infer the names. I dont see the benefits of a generic logic like this which is not intuitive if we can have other alternatives.

How about defining regular expressions in the individual serializers and using them in the buildQueue/buildTopic/buildSubscription methods?

Reply:
Since the pattern is same for across entities, it seems valuable to have it be defined in a common way.
Moreover, this needs to be repeated on individual `entry` instances when dealing with `feed`.